addons/point_of_sale/static/src/js/models.js

```diff
@@ -655,33 +655,35 @@ exports.PosModel = Backbone.Model.extend({
         this.set('cashier', user);
         this.db.set_cashier(this.get('cashier'));
     },
-    //creates a new empty order and sets it as the current order
+    // creates a new empty order and sets it as the current order
     add_new_order: function(){
         var order = new exports.Order({},{pos:this});
         this.get('orders').add(order);
         this.set('selectedOrder', order);
         return order;
     },
-    // load the locally saved unpaid orders for this session.
+    // load the locally saved unpaid orders for this PoS Config.
     load_orders: function(){
         var jsons = this.db.get_unpaid_orders();
         var orders = [];
         var not_loaded_count = 0;
 
         for (var i = 0; i < jsons.length; i++) {
             var json = jsons[i];
-            if (json.pos_session_id === this.pos_session.id) {
-                orders.push(new exports.Order({},{
-                    pos:  this,
-                    json: json,
-                }));
+            if (json.config_id === this.pos_session.config_id[0]) {
+                if (json.lines.length > 0) {
+                    orders.push(new exports.Order({},{
+                        pos:  this,
+                        json: json,
+                    }));
+                }
             } else {
                 not_loaded_count += 1;
```

Comment:
I think this part just became completely useless since the way localstorage is used makes it impossible to have orders belonging to another config in the localstorage currently in use - the session storage is uniquified by the pos_config uuid, making a mixup like this impossible (via models.js#255)